### ISSUE TITLE
CFF not carrying forward reports correctly

### ISSUE THREAD

**Issue by rohan-at-sentry** [codecov/engineering-team#2250]
**Describe the bug**
CFF not carrying forward reports correctly

**To Reproduce**
https://github.com/getsentry/sentry/pull/75842

**Expected behavior**
Reports are carried forward correctly

**Additional context**
Causes https://github.com/getsentry/sentry/pull/75842. The getsentry/sentry repo is configured to use CFF, after_n_builds and failed test reporting in addition to coverage reporting. For https://github.com/getsentry/sentry/pull/75842, we didn't update the failed test comment (after all tests passed) because of a not enough builds error (if after_n_builds is turned on, and the # of uploads <n) . This happened because some reports were not being carried forward correctly.

We became aware of the issue because the failed test comment persisted even after all tests were subsequently passing. What's worrying is this may be happening silently for those customers that don't have the failed test comment turned on and all they may be seeing is PR comment with coverage that is wrong. We haven't had customers call in with unexplained coverage changes, but it'll be helpful to diagnose what exactly is going on with CFF and after_n_builds and ship a fix ASAP. This is being tracked in https://github.com/codecov/engineering-team/issues/2250.

**Comment by rohan-at-sentry** [comment]
Part of the problem may be fixed by doing https://github.com/codecov/engineering-team/issues/1245.

**Comment by joseph-sentry** [comment]
tldr: I broke Sentry coverage and test results upload from main. `after_n_builds` was working correctly, CFF is working as it should though we should determine if this is the desired behaviour for CFF.

1. I broke the Codecov uploads from main by setting the commit_sha to be an empty string
2. This made it so all coverage and test results uploads from the main branch were failing
3. However, bundle uploads were still working fine, so the commits were existent in our DB
4. So those commits existed in our DB, but had no coverage uploads whatsoever
5. So when PRs would be made with those commits as the base we would carry forward nothing because the commit existed and it had no coverage uploads.

I guess a solution to this would be carrying forward uploads on commit creation, or moving uploads to the merge commit on merge (not sure if this is feasible)
